File: 3. Herencia, extension e implementacion/En Java/hei/src/HEI.java
import model.ICrystalGlass;
import model.IVaso;
import model.VasoDeChupitoDeCristal;

public class HEI {
    public static void main(String[] args) throws Exception {
        System.out.println();
        IVaso vaso = new VasoDeChupitoDeCristal(0.01, 0.30);
        System.out.println("Capacidad: " + vaso.getCapacity());
        System.out.println("Peso: " + vaso.getWeight());
        if(vaso instanceof ICrystalGlass) System.out.println("El vaso es de cristal");
        else System.err.println("El vaso no es de cristal");
    }
}

File: 3. Herencia, extension e implementacion/En C#/HEI.cs
using System;

namespace model
{
    class HEI
    {
        static void Main(string[] args)
        {
            IVaso vaso = new VasoDeChupitoDeCristal(0.01, 0.30);
            Console.WriteLine("Capacidad: " + vaso.getCapacity());
            Console.WriteLine("Peso: " + vaso.getWeight());
            if(vaso is ICrystalGlass) Console.WriteLine("El vaso es de cristal");
            else Console.WriteLine("El vaso no es de cristal");
        }
    }
}
